### DISCUSSION TITLE
Reconcile function fail and move on

### DISCUSSION THREAD

**Discussion by jakebajo** [Nixtla/hierarchicalforecast#47] [Ideas]
Is there a way to write the `HierarchicalReconciliation.reconcile()` function so that if one reconcile method fails, the function will move on to the next one and still output a dataframe with the reconcile methods that worked?

I understand that you can loop through various reconcile methods instead but that is slower and inefficient. I don't think this would be a huge change in the function, just some try except essentially to the for loop. Could also output an error dataframe if necessary. I could possibly put some things together in a pull request if I have time but wanted to ask if the team has thought about this.

**Comment by kdgutier** [comment]
Hey @jakebajo,

Thanks for the suggestion, we have done a similar approach in the past to protect the convergence of `AutoARIMA`.
If you are using methods that require an inverse or a pseudo-inverse I recommend you to explore the regularization parameters that aim to make the reconciliation numerically stable, for example MinT-ridge and ERM-lasso.

If you want to take a look to protect the reconciliation with try and excepts, your effort will be welcome.
